go/vt/vttablet/tabletserver/connpool/pool.go

```diff
@@ -160,6 +162,7 @@ func (cp *Pool) Get(ctx context.Context) (*DBConn, error) {
 		waiterCount := cp.waiterCount.Add(1)
 		defer cp.waiterCount.Add(-1)
 		if waiterCount > cp.waiterCap {
+			cp.waiterQueueFull.Add(1)
 			return nil, vterrors.Errorf(vtrpcpb.Code_RESOURCE_EXHAUSTED, "pool %s waiter count exceeded", cp.name)
 		}
 	}
```

Comment:
I noticed that this new metric won't be added to the `StatsJSON` output -- does that matter?

Reply:
Can you add it here, please?

Reply:
I'm not sure that's possible the way things are architected currently. In the code you linked, we only have access to the ResourcePool struct. But the metric we are adding here is only present on the Pool struct.

Reply:
Here's one potential idea: https://gist.github.com/mattlord/b305eabac4ab14c913fc9796538319d5

Reply:
Here's another rough idea: https://gist.github.com/mattlord/01dbb1c52c8c2fc84a103de1869e82df

Reply:
Here's another rough idea: https://gist.github.com/mattlord/684656143768d692b71402ff796e98c0

Reply:
Perhaps someone else has an even better idea 🙂

Reply:
I agree with @mattlord that we should include them in the `StatsJSON` function. The easiest and IMO the cleanest solution is to just append to the output of the underlying call here before returning https://github.com/vitessio/vitess/blob/904c9a956028d58e5c4f666dfeaa23042ef482cf/go/vt/vttablet/tabletserver/connpool/pool.go#L233-L239

So it reads like -
```go
func (cp *Pool) StatsJSON() string { 
 	p := cp.pool() 
 	if p == nil { 
 		return "{}" 
 	} 
 	res:= p.StatsJSON()
 	res += fmt.Sprintf(`"WaiterQueueFull": %v`, p.waiterQueueFull.Get())
 	return res
}
```